### ISSUE TITLE
`nn.CrossEntropyLoss` accepts negative target probabilities

### ISSUE THREAD

**Issue by meilame-tayebjee** [pytorch/pytorch#152437]
### 📚 The doc issue

The `CrossEntropyLoss` [documentation](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) mentions that: 

> **Target**:  
> - If containing **class indices**, the shape should be:
>   - `()` (scalar),
>   - `(N)`, or
>   - `(N, d1, d2, ..., dK)` with `K ≥ 1` (for K-dimensional loss),
>   where each value should be in the range `[0, C)`.  
>   The target data type is required to be `long` (i.e., `torch.int64`) when using class indices.
> 
> - If containing **class probabilities**, the target must have the **same shape as the input**, and each value should be in the range `[0, 1]`.  
>   This means the target data type is required to be `float` (e.g., `torch.float32`) when using class probabilities.

In the second case (when passing target as _probabilities_), nothing seems to enforce that each value should be in the range `[0, 1]`. The user can even pass "negative probabilities", for instance:

```
import torch 
import torch.nn as nn

num_classes = 5
batch_size = 10
loss = nn.CrossEntropyLoss(reduction='none')

input = 2*torch.randn(batch_size, num_classes, requires_grad=True)
target = (-2) * torch.ones(batch_size, num_classes)
output = loss(input, target)
output
```

Using version 2.6.0, this code runs without throwing any error or warning.

It would be beneficial to either document this non-enforcement (so that the user keeps in mind they have to do it themselves) or to enforce it.

**Comment by albanD** [comment]
Doing such a check would be expensive since we don't have a custom kernel right now for CrossEntropyLoss, that's why it doesn't exist.
We would need to add such a kernel to be able to do error checking without a major perf hit :/
